src/socketio.quiver.js

```diff
@@ -1390,6 +1404,41 @@ QuiverSocialProvider.prototype.err = function(code) {
   return err;
 };
 
+/*
+ * Initialize this.logger_.
+ * @private
+ */
+QuiverSocialProvider.prototype.initLogger_ = function() {
+  this.logger_ = console;  // Initialize to console if it exists.
+  if (typeof freedom !== 'undefined' &&
+      typeof freedom.core === 'function') {
+    freedom.core().getLogger('[QuiverSocialProvider]').then(function(log) {
+      this.logger_ = log;
```

Comment:
Looks like `this.logger_` is undefined until this promise returns, which means that `this.log` et. al. will throw if called too early.  Seems dangerous.